### PR TITLE
WIP: Adding a group tab

### PR DESCRIPTION
**Pull request by RoySegall**
#207

### CODE REVIEW

File: og_ui/og_ui.module [hunk 1]
```diff
@@ -95,3 +96,26 @@ function og_ui_entity_type_save(EntityInterface $entity) {
...
+  
+  /** @var $entity_types \Drupal\Core\Entity\EntityTypeInterface[] */
+  foreach ($entity_types as $entity_type_id => $entity_type) {
+
+//    if (!$group_manager->getGroupsForEntityType($entity_type_id)) {
+//      continue;
+//    }
+
+    // Adding the group manager causing for a circular dependency because the
+    // group manager the the entity manager as a dependency.
```
Comment (RoySegall) [inline og_ui/og_ui.module:123]: @pfrenssen @damiankloip Any ideas over here? As far as I can see it we need to remove part of the logic that determine if this is a group or not outside the group manager.

**Comment by RoySegall** [comment]
First sketch:
tasks___site-install

File: og_ui/src/OgUiAdminRouteAbstract.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Drupal\og_ui;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Plugin\PluginBase;
+
+abstract class OgUiAdminRouteAbstract extends PluginBase implements OgUiAdminRouteInterface {
```
Comment (amitaibu) [inline og_ui/src/OgUiAdminRouteAbstract.php]: I think it's usually `OgUiAdminRouteAbstract` => `OgUiAdminRouteBase`

**Comment by RoySegall** [comment]
I burn some free time on the flight. Looks good: tasks___site-install

File: og_ui/src/Access/OgUiRoutingAccess.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\og\Og;
+
+/**
+ * Checks access for displaying configuration translation page.
+ */
+class OgUiRoutingAccess implements AccessInterface {
+
```
Comment (RoySegall) [inline og_ui/src/Access/OgUiRoutingAccess.php]: Remove this line.

File: og_ui/src/Controller/PeopleController.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+use Drupal\Core\Session\SessionManager;
+use Drupal\Core\Url;
+use Drupal\og_ui\Form\OgUiAddPeopleForm;
+use Drupal\og_ui\OgUi;
+use Drupal\og_ui\OgUiAdminRouteInterface;
+use Drupal\user\PrivateTempStoreFactory;
+use Drupal\views\Views;
+
+class PeopleController extends ControllerBase {
+
```
Comment (RoySegall) [inline og_ui/src/Controller/PeopleController.php]: Remove extra line.

File: og_ui/src/Controller/PeopleController.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  /**
+   * Add people to the group.
+   */
+  public function addPeopleForm() {
+    return \Drupal::formBuilder()->getForm('\Drupal\og_ui\Form\OgUiAddPeopleForm');
+  }
+
+}
+
```
Comment (RoySegall) [inline og_ui/src/Controller/PeopleController.php]: Remove extra line.

File: og_ui/src/Form/DeleteMultiple.php [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+use Drupal\og\Entity\OgMembership;
+use Drupal\user\PrivateTempStoreFactory;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a og_ui deletion confirmation form.
+ */
+class DeleteMultiple extends ConfirmFormBase {
+  /**
```
Comment (RoySegall) [inline og_ui/src/Form/DeleteMultiple.php:25]: Add an extra line.

File: og_ui/src/Form/OgUiAddPeopleForm.php [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    OgMembership::create([
+      'type' => OgMembership::TYPE_DEFAULT,
+      'uid' => $form_state->getValue('username'),
+      'entity_type' => OgUi::getEntity()->getEntityTypeId(),
+      'entity_id' => OgUi::getEntity()->id(),
+    ])->save();
+
+//    $form_state->setRedirectUrl(Url::fromUserInput($this->tempStoreFactory->get('og_ui')->get('people_url')));
```
Comment (RoySegall) [inline og_ui/src/Form/OgUiAddPeopleForm.php]: Store the address on the temp storage.

File: og_ui/src/Plugin/GroupAdminRoutes/People.php [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+ * )
+ */
+class People extends OgUiAdminRouteAbstract {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access() {
+    return AccessResultAllowed::allowedIf(TRUE);
+//    return AccessResultAllowed::allowedIf($this->getMembership()->hasPermission('administer group'));
```
Comment (RoySegall) [inline og_ui/src/Plugin/GroupAdminRoutes/People.php]: Handle permission.

**Comment by RoySegall** [comment]
@amitaibu people___site-install

**Comment by amitaibu** [comment]
Nice!

File: og_ui/src/Controller/PeopleController.php [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      ->setGroup(OgUi::getEntity())
+      ->access();
+  }
+
+  /**
+   * Display list of people which belong to the group.
+   */
+  public function PeopleList() {
+    /** @var PrivateTempStoreFactory $session_storage */
+    $session_storage = \Drupal::service('user.private_tempstore');
```
Comment (amitaibu) [inline og_ui/src/Controller/PeopleController.php:20]: What is this for?

Comment (RoySegall) [inline og_ui/src/Controller/PeopleController.php:20]: Used to store the last route we been in order to redirect the user back to where he came from.

We did the same thing in the message module.

**Comment by RoySegall** [comment]
OK. so we got adding members, deleting members.

There couples of weird stuff as:
1. redirecting after submitting the form
2. Getting the group content from the menu with out loading.
3. Handle circular dependancies in `og_ui_entity_type_alter()`

File: og_ui/src/Controller/PeopleController.php (hunk 9 above)
Comment (amitaibu) [inline og_ui/src/Controller/PeopleController.php:20]: Then we should remove it from there as well ;)

We can have the current route as a hidden value in the form instead,  no?

**Comment by RoySegall** [comment]
@amitaibu I handled the url redirects and getting the group from the context.

File: og_ui/src/Controller/PeopleController.php (hunk 9 above)
Comment (RoySegall) [inline og_ui/src/Controller/PeopleController.php:20]: The way core implements multiple delete of items are without any confirm form. Just deleting and that's it.

If so we can remove the form(which also need to impact Message). For now I remove the redirect to the form.

**Comment by RoySegall** [comment]
I keep with covering the tests but there are two different things:
1. The tab people does not appear in the page although i have access to the page
2. When visiting the users in the group i get nothing.

File: og_ui/src/OgUi.php [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+  /**
+   * Get the current entity from the route.
+   *
+   * @return \Drupal\Core\Entity\ContentEntityBase
+   */
+  public static function getEntity() {
+    // Un used for unit testing due to the fact that in unit testing the route
+    // options does not contain the current viewed entity.
+    // todo: find a better solution.
```
Comment (RoySegall) [inline og_ui/src/OgUi.php:37]: @amitaibu ^^

Thoughts on how we can pass it?

**Comment by RoySegall** [comment]
@pfrenssen Are around here? Can you have a look on this one?

There is something very strange in this one. The view don't display the users in the group although i'm getting the group context, I checked this yesterday. I just don't know why I can't see the list of users in the group.

**Comment by amitaibu** [comment]
What's the status here? Seems a lot of work already done - how far are we? :)

**Comment by RoySegall** [comment]
For some reason the view don't display the list of people when i'm in unit testing. I'll try to tackle this one again.

**Comment by RoySegall** [comment]
I got back to this one. I merged master and now the sniffer gone mad but now when I'm looking in the view i can see that the demo user is listed, something that wasnt there before:
people___drupal

The other problem is when I'm searching for the user and i'm getting:

> Drupal\Tests\og_ui\Tests\Functional\PeopleTabTest::testPeopleTab
> simplexml_import_dom(): Invalid Nodetype to import

I think I'll branch out from the master and them continue the work.

**Comment by amitaibu** [comment]
Ok, I can have a look at this, while you work on OgContext 😉

**Comment by amitaibu** [comment]
I'm going to break this PR down to smaller PRs, so it's easier to add tests and review.

**Comment by amitaibu** [comment]
Closed in favor of https://github.com/Gizra/og/pull/149

I'm not deleting this branch, as https://github.com/Gizra/og/pull/149 will not keep all the files in here (such as the people view, add member, delete member, etc') to reduce the PR scope.

**Pull request closed without merging**